### PR TITLE
Clean up isinstance checks for tuples | feat(torchlib)

### PR DESCRIPTION
**Pull request by justinchuby**
Clean up isinstance checks for tuples to turn some trace_only functions into script functions. This is done by updating the tracing evaluator to turn int attributes into sequences according to the function op schema.

Remove xfails for var_mean.

Fixes https://github.com/microsoft/onnxscript/issues/1191

### CODE REVIEW

**Comment by justinchuby** [comment]
Windows: Use the helper to cast to int64

File: onnxscript/function_libs/torch_lib/graph_building.py [hunk 1]
```diff
@@ -368,13 +368,22 @@ def eval_function(  # type: ignore[override]
...
-            param = name_to_schema[name]
+            attribute = op_schema.attributes[name]
             # Cast int to float if needed
-            if param.type in {float, "float"}:
-                # FIXME(justinchuby): Create invariant on the type of param.type to simplify this
+            if attribute.type == onnx.defs.OpSchema.AttrType.FLOAT:
+                # Cast int to float is the attribute is FLOAT
                 attributes[name] = float(value)
+            if attribute.type == onnx.defs.OpSchema.AttrType.INTS and isinstance(value, int):
+                attributes[name] = (value,)
```
Comment (BowenBao) [inline onnxscript/function_libs/torch_lib/graph_building.py:382]: when would this happen?

**Review by BowenBao — APPROVED** [review]
LGTM w/ inline questions.

File: onnxscript/function_libs/torch_lib/graph_building.py [hunk 2]
```diff
@@ -368,13 +368,22 @@ def eval_function(  # type: ignore[override]
...
         )
-        name_to_schema = {param.name: param for param in param_schemas}
+
+        # Cast attributes to the correct type based on function signature
+        op_schema = function.op_schema
+        assert op_schema is not None
         for name, value in attributes.items():
-            param = name_to_schema[name]
+            attribute = op_schema.attributes[name]
             # Cast int to float if needed
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/graph_building.py:377]:
```suggestion
```

File: onnxscript/function_libs/torch_lib/graph_building.py (hunk 1 above)
Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/graph_building.py:381]: In PyTorch, attributes annotated as `int[1]?` accepts an integer or a sequence. When the attribute is an integer, it is treated as a single element sequence. ONNX requires an attribute to either be an integer or a sequence. So we promote the value to a sequence here.

Added as a comment too.

**Pull request merged**